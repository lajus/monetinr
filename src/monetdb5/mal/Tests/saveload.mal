# save/load tests

a:= bat.new(:oid,:int);
bat.append(a,123);
bat.append(a,321);
bat.setName(a,"aaa");
io.print(a);

b:= bat.new(:oid,:str);
bat.insert(b,0@0,"hello");
bat.insert(b,1@0,"world");
bat.setName(b,"bbb");
io.print(b);

bat.save(a);
bat.save(b);

ar:bat[:oid,:int]:= bat.load("aaa");
io.print(ar);
br:bat[:oid,:str]:= bat.load("bbb");
io.print(br);


#and an error
er:bat[:oid,:int]:= bbp.bind("bbb");
io.print(er);
