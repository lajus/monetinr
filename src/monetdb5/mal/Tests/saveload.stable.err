stderr of test 'saveload` in directory 'monetdb5/mal` itself:


# 20:31:51 >  
# 20:31:51 >   mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs"  --set mapi_open=true --set xrpc_open=true --set mapi_port=35708 --set xrpc_port=42614 --set monet_prompt= --trace  --dbname=mTests_src_mal  saveload.mal
# 20:31:51 >  

#warning: please don't forget to set your vault key!
#(see /ufs/mk/monet5/Linux/etc/monetdb5.conf)

# 20:31:51 >  
# 20:31:51 >  Done.
# 20:31:51 >  

