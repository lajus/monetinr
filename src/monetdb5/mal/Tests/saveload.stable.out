stdout of test 'saveload` in directory 'monetdb5/mal` itself:


# 20:31:51 >  
# 20:31:51 >   mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs"  --set mapi_open=true --set xrpc_open=true --set mapi_port=35708 --set xrpc_port=42614 --set monet_prompt= --trace  --dbname=mTests_src_mal  saveload.mal
# 20:31:51 >  

# MonetDB server v5.5.0, based on kernel v1.23.0
# Serving database 'mTests_src_mal'
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-2008 CWI, all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://gio.ins.cwi.nl:35708/
## save/load tests
#
#a:= bat.new(:int,:int);
#bat.insert(a,0,123);
#bat.insert(a,0,321);
#bat.setName(a,"aaa");
#io.print(a);
#
#b:= bat.new(:oid,:str);
#bat.insert(b,0@0,"hello");
#bat.insert(b,1@0,"world");
#bat.setName(b,"bbb");
#io.print(b);
#
#bat.save(a);
#bat.save(b);
#
#ar:bat[:int,:int]:= bat.load("aaa");
#io.print(ar);
#br:bat[:oid,:str]:= bat.load("bbb");
#io.print(br);
#
#
##and an error
#er:bat[:int,:int]:= bbp.bind("bbb");
#io.print(er);
#
function user.main():void;
# save/load tests 
    a := bat.new(:oid,:int);
    bat.append(a,123);
    bat.append(a,321);
    bat.setName(a,"aaa");
    io.print(a);
    b := bat.new(:oid,:str);
    bat.insert(b,0@0,"hello");
    bat.insert(b,1@0,"world");
    bat.setName(b,"bbb");
    io.print(b);
    bat.save(a);
    bat.save(b);
    ar:bat[:oid,:int]  := bat.load("aaa");
    io.print(ar);
    br:bat[:oid,:str]  := bat.load("bbb");
    io.print(br);
#and an error 
    er:bat[:oid,:int]  := bbp.bind("bbb");
    io.print(er);
end main;
#-----------------#
# h	t	  # name
# int	int	  # type
#-----------------#
[ 0@0,	  123	  ]
[ 1@0,	  321	  ]
#-------------------------#
# h	t		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "world"	  ]
#-----------------#
# h	t	  # name
# int	int	  # type
#-----------------#
[ 0@0,	  123	  ]
[ 1@0,	  321	  ]
#-------------------------#
# h	t		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "world"	  ]
!MALException:bbp.bind:Type mismatch





# 20:31:51 >  
# 20:31:51 >  Done.
# 20:31:51 >  

